plugins/kubernetes-backend/src/cluster-locator/MultiTenantConfigClusterLocator.ts

```diff
@@ -28,20 +29,45 @@ export class MultiTenantConfigClusterLocator
   }
 
   static fromConfig(config: Config[]): MultiTenantConfigClusterLocator {
+    // TODO: Add validation that authProvider is required and serviceAccountToken
+    // is required if authProvider is serviceAccount
     return new MultiTenantConfigClusterLocator(
       config.map(c => {
         return {
           name: c.getString('name'),
           url: c.getString('url'),
           serviceAccountToken: c.getOptionalString('serviceAccountToken'),
+          authProvider: (c.getOptionalString('authProvider')
+            ? c.getOptionalString('authProvider')
```

Comment:
Just `c.getOptionalString('authProvider') ?? 'serviceAccount'` here. I'd also add validation that it's actually a `AuthProviderType` at runtime instead of casting too